In [1]:
import pandas as pd
import numpy as np

In [7]:
from google.colab import files
uploaded=files.upload()

Saving SalaryData_Test.csv to SalaryData_Test.csv


In [8]:
salarydata_test=pd.read_csv("SalaryData_Test.csv")
salarydata_test

age      workclass      education  educationno        maritalstatus  \
0       25        Private           11th            7        Never-married   
1       38        Private        HS-grad            9   Married-civ-spouse   
2       28      Local-gov     Assoc-acdm           12   Married-civ-spouse   
3       44        Private   Some-college           10   Married-civ-spouse   
4       34        Private           10th            6        Never-married   
...    ...            ...            ...          ...                  ...   
15055   33        Private      Bachelors           13        Never-married   
15056   39        Private      Bachelors           13             Divorced   
15057   38        Private      Bachelors           13   Married-civ-spouse   
15058   44        Private      Bachelors           13             Divorced   
15059   35   Self-emp-inc      Bachelors           13   Married-civ-spouse   

               occupation    relationship                 race      sex  \
0       Machine-op-inspct       Own-child                Black     Male   
1         Farming-fishing         Husband                White     Male   
2         Protective-serv         Husband                White     Male   
3       Machine-op-inspct         Husband                Black     Male   
4           Other-service   Not-in-family                White     Male   
...                   ...             ...                  ...      ...   
15055      Prof-specialty       Own-child                White     Male   
15056      Prof-specialty   Not-in-family                White   Female   
15057      Prof-specialty         Husband                White     Male   
15058        Adm-clerical       Own-child   Asian-Pac-Islander     Male   
15059     Exec-managerial         Husband                White     Male   

       capitalgain  capitalloss  hoursperweek          native  Salary  
0                0            0            40   United-States   <=50K  
1                0            0            50   United-States   <=50K  
2                0            0            40   United-States    >50K  
3             7688            0            40   United-States    >50K  
4                0            0            30   United-States   <=50K  
...            ...          ...           ...             ...     ...  
15055            0            0            40   United-States   <=50K  
15056            0            0            36   United-States   <=50K  
15057            0            0            50   United-States   <=50K  
15058         5455            0            40   United-States   <=50K  
15059            0            0            60   United-States    >50K  

[15060 rows x 14 columns]

In [9]:
from sklearn import preprocessing
label_encoder = preprocessing.LabelEncoder()
salarydata_test['workclass']= label_encoder.fit_transform(salarydata_test['workclass']) 
salarydata_test['education']= label_encoder.fit_transform(salarydata_test['education'])
salarydata_test['maritalstatus']= label_encoder.fit_transform(salarydata_test['maritalstatus'])
salarydata_test['occupation']= label_encoder.fit_transform(salarydata_test['occupation'])
salarydata_test['relationship']= label_encoder.fit_transform(salarydata_test['relationship'])
salarydata_test['race']= label_encoder.fit_transform(salarydata_test['race'])
salarydata_test['sex']= label_encoder.fit_transform(salarydata_test['sex'])
salarydata_test['native']= label_encoder.fit_transform(salarydata_test['native'])
salarydata_test['Salary']= label_encoder.fit_transform(salarydata_test['Salary'])

In [10]:
from google.colab import files 
uploaded=files.upload()

Saving SalaryData_Train.csv to SalaryData_Train.csv


In [11]:
salarydata_train=pd.read_csv("SalaryData_Train.csv")
salarydata_train

age          workclass    education  educationno        maritalstatus  \
0       39          State-gov    Bachelors           13        Never-married   
1       50   Self-emp-not-inc    Bachelors           13   Married-civ-spouse   
2       38            Private      HS-grad            9             Divorced   
3       53            Private         11th            7   Married-civ-spouse   
4       28            Private    Bachelors           13   Married-civ-spouse   
...    ...                ...          ...          ...                  ...   
30156   27            Private   Assoc-acdm           12   Married-civ-spouse   
30157   40            Private      HS-grad            9   Married-civ-spouse   
30158   58            Private      HS-grad            9              Widowed   
30159   22            Private      HS-grad            9        Never-married   
30160   52       Self-emp-inc      HS-grad            9   Married-civ-spouse   

               occupation    relationship    race      sex  capitalgain  \
0            Adm-clerical   Not-in-family   White     Male         2174   
1         Exec-managerial         Husband   White     Male            0   
2       Handlers-cleaners   Not-in-family   White     Male            0   
3       Handlers-cleaners         Husband   Black     Male            0   
4          Prof-specialty            Wife   Black   Female            0   
...                   ...             ...     ...      ...          ...   
30156        Tech-support            Wife   White   Female            0   
30157   Machine-op-inspct         Husband   White     Male            0   
30158        Adm-clerical       Unmarried   White   Female            0   
30159        Adm-clerical       Own-child   White     Male            0   
30160     Exec-managerial            Wife   White   Female        15024   

       capitalloss  hoursperweek          native  Salary  
0                0            40   United-States   <=50K  
1                0            13   United-States   <=50K  
2                0            40   United-States   <=50K  
3                0            40   United-States   <=50K  
4                0            40            Cuba   <=50K  
...            ...           ...             ...     ...  
30156            0            38   United-States   <=50K  
30157            0            40   United-States    >50K  
30158            0            40   United-States   <=50K  
30159            0            20   United-States   <=50K  
30160            0            40   United-States    >50K  

[30161 rows x 14 columns]

In [12]:
from sklearn import preprocessing
label_encoder = preprocessing.LabelEncoder()
salarydata_train['workclass']= label_encoder.fit_transform(salarydata_train['workclass']) 
salarydata_train['education']= label_encoder.fit_transform(salarydata_train['education'])
salarydata_train['maritalstatus']= label_encoder.fit_transform(salarydata_train['maritalstatus'])
salarydata_train['occupation']= label_encoder.fit_transform(salarydata_train['occupation'])
salarydata_train['relationship']= label_encoder.fit_transform(salarydata_train['relationship'])
salarydata_train['race']= label_encoder.fit_transform(salarydata_train['race'])
salarydata_train['sex']= label_encoder.fit_transform(salarydata_train['sex'])
salarydata_train['native']= label_encoder.fit_transform(salarydata_train['native'])
salarydata_train['Salary']= label_encoder.fit_transform(salarydata_train['Salary'])

In [13]:
salarydata_test.head()


age  workclass  education  educationno  maritalstatus  occupation  \
0   25          2          1            7              4           6   
1   38          2         11            9              2           4   
2   28          1          7           12              2          10   
3   44          2         15           10              2           6   
4   34          2          0            6              4           7   

   relationship  race  sex  capitalgain  capitalloss  hoursperweek  native  \
0             3     2    1            0            0            40      37   
1             0     4    1            0            0            50      37   
2             0     4    1            0            0            40      37   
3             0     2    1         7688            0            40      37   
4             1     4    1            0            0            30      37   

   Salary  
0       0  
1       0  
2       1  
3       1  
4       0

In [14]:
salarydata_train.head()

age  workclass  education  educationno  maritalstatus  occupation  \
0   39          5          9           13              4           0   
1   50          4          9           13              2           3   
2   38          2         11            9              0           5   
3   53          2          1            7              2           5   
4   28          2          9           13              2           9   

   relationship  race  sex  capitalgain  capitalloss  hoursperweek  native  \
0             1     4    1         2174            0            40      37   
1             0     4    1            0            0            13      37   
2             1     4    1            0            0            40      37   
3             0     2    1            0            0            40      37   
4             5     2    0            0            0            40       4   

   Salary  
0       0  
1       0  
2       0  
3       0  
4       0

In [15]:
salarydata_test.shape

(15060, 14)

In [16]:
salarydata_train.shape

(30161, 14)

In [17]:
salarydata_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15060 entries, 0 to 15059
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype
---  ------         --------------  -----
 0   age            15060 non-null  int64
 1   workclass      15060 non-null  int64
 2   education      15060 non-null  int64
 3   educationno    15060 non-null  int64
 4   maritalstatus  15060 non-null  int64
 5   occupation     15060 non-null  int64
 6   relationship   15060 non-null  int64
 7   race           15060 non-null  int64
 8   sex            15060 non-null  int64
 9   capitalgain    15060 non-null  int64
 10  capitalloss    15060 non-null  int64
 11  hoursperweek   15060 non-null  int64
 12  native         15060 non-null  int64
 13  Salary         15060 non-null  int64
dtypes: int64(14)
memory usage: 1.6 MB


In [18]:
salarydata_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30161 entries, 0 to 30160
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype
---  ------         --------------  -----
 0   age            30161 non-null  int64
 1   workclass      30161 non-null  int64
 2   education      30161 non-null  int64
 3   educationno    30161 non-null  int64
 4   maritalstatus  30161 non-null  int64
 5   occupation     30161 non-null  int64
 6   relationship   30161 non-null  int64
 7   race           30161 non-null  int64
 8   sex            30161 non-null  int64
 9   capitalgain    30161 non-null  int64
 10  capitalloss    30161 non-null  int64
 11  hoursperweek   30161 non-null  int64
 12  native         30161 non-null  int64
 13  Salary         30161 non-null  int64
dtypes: int64(14)
memory usage: 3.2 MB


In [19]:
salarydata_train.describe()

age    workclass    education   educationno  maritalstatus  \
count  30161.000000  30161.00000  30161.00000  30161.000000   30161.000000   
mean      38.438115      2.19933     10.33361     10.121316       2.580087   
std       13.134830      0.95394      3.81226      2.550037       1.498018   
min       17.000000      0.00000      0.00000      1.000000       0.000000   
25%       28.000000      2.00000      9.00000      9.000000       2.000000   
50%       37.000000      2.00000     11.00000     10.000000       2.000000   
75%       47.000000      2.00000     12.00000     13.000000       4.000000   
max       90.000000      6.00000     15.00000     16.000000       6.000000   

         occupation  relationship          race           sex   capitalgain  \
count  30161.000000  30161.000000  30161.000000  30161.000000  30161.000000   
mean       5.959849      1.418322      3.678592      0.675707   1092.044064   
std        4.029633      1.601361      0.834721      0.468118   7406.466611   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        2.000000      0.000000      4.000000      0.000000      0.000000   
50%        6.000000      1.000000      4.000000      1.000000      0.000000   
75%        9.000000      3.000000      4.000000      1.000000      0.000000   
max       13.000000      5.000000      4.000000      1.000000  99999.000000   

        capitalloss  hoursperweek        native        Salary  
count  30161.000000  30161.000000  30161.000000  30161.000000  
mean      88.302311     40.931269     35.413481      0.248931  
std      404.121321     11.980182      5.956430      0.432401  
min        0.000000      1.000000      0.000000      0.000000  
25%        0.000000     40.000000     37.000000      0.000000  
50%        0.000000     40.000000     37.000000      0.000000  
75%        0.000000     45.000000     37.000000      0.000000  
max     4356.000000     99.000000     39.000000      1.000000

In [20]:
salarydata_test.describe()

age     workclass     education   educationno  maritalstatus  \
count  15060.000000  15060.000000  15060.000000  15060.000000   15060.000000   
mean      38.768327      2.214874     10.272377     10.112749       2.595286   
std       13.380676      0.966422      3.826251      2.558727       1.505336   
min       17.000000      0.000000      0.000000      1.000000       0.000000   
25%       28.000000      2.000000      9.000000      9.000000       2.000000   
50%       37.000000      2.000000     11.000000     10.000000       2.000000   
75%       48.000000      2.000000     12.000000     13.000000       4.000000   
max       90.000000      6.000000     15.000000     16.000000       6.000000   

         occupation  relationship          race           sex   capitalgain  \
count  15060.000000  15060.000000  15060.000000  15060.000000  15060.000000   
mean       5.989044      1.401394      3.683665      0.673772   1120.301594   
std        4.020113      1.588954      0.827630      0.468848   7703.181842   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        2.000000      0.000000      4.000000      0.000000      0.000000   
50%        6.000000      1.000000      4.000000      1.000000      0.000000   
75%        9.000000      3.000000      4.000000      1.000000      0.000000   
max       13.000000      5.000000      4.000000      1.000000  99999.000000   

        capitalloss  hoursperweek        native        Salary  
count  15060.000000  15060.000000  15060.000000  15060.000000  
mean      89.041899     40.951594     35.467596      0.245684  
std      406.283245     12.062831      5.880922      0.430506  
min        0.000000      1.000000      0.000000      0.000000  
25%        0.000000     40.000000     37.000000      0.000000  
50%        0.000000     40.000000     37.000000      0.000000  
75%        0.000000     45.000000     37.000000      0.000000  
max     3770.000000     99.000000     39.000000      1.000000

In [21]:
salarydata_test.isna().sum()

age              0
workclass        0
education        0
educationno      0
maritalstatus    0
occupation       0
relationship     0
race             0
sex              0
capitalgain      0
capitalloss      0
hoursperweek     0
native           0
Salary           0
dtype: int64

In [22]:
salarydata_train.isna().sum()

age              0
workclass        0
education        0
educationno      0
maritalstatus    0
occupation       0
relationship     0
race             0
sex              0
capitalgain      0
capitalloss      0
hoursperweek     0
native           0
Salary           0
dtype: int64

In [23]:
#Applying train and split test on salary dataset 
x_train=salarydata_train.iloc[:,0:13]
y_train=salarydata_train.iloc[:,13]  


In [24]:
x_test = salarydata_test.iloc[:,0:13]
y_test = salarydata_test.iloc[:,13]

Applying NAIVE BAYES

In [25]:
#Prepare a Multinomial Naive Bayes model on training dataset
from sklearn.naive_bayes import  MultinomialNB as MB      

In [26]:
classifier_mb=MB()


In [27]:
classifier_mb.fit(x_train,y_train)

MultinomialNB()

In [28]:
#Model accuracy on train set
train_pred_m = classifier_mb.predict(x_train)
accuracy_train_m = np.mean(train_pred_m==y_train)

In [29]:
#Model Accuracy on test set
test_pred_m = classifier_mb.predict(x_train)
accuracy_test_m = np.mean(test_pred_m==y_train)

In [30]:
accuracy_train_m

0.7729186698053778

In [31]:
accuracy_test_m

0.7729186698053778

In [32]:
#We got 77.29% of accuracy after implementation of Multinomial Naive Bayes 

**2. Categorical NB**

In [33]:
#Preparing a Categorical naive bayes on training data set 
from sklearn.naive_bayes import CategoricalNB as CNB

In [34]:
classifier_cnb=CNB()
classifier_cnb.fit(x_train,y_train)

CategoricalNB()

In [35]:
#Model Acurracy on train set 
train_pred_cnb=classifier_cnb.predict(x_train)
accuracy_train_cnb=np.mean(train_pred_cnb==y_train)

In [36]:
accuracy_train_cnb

0.8571002287722556

APPLYING GAUSSIAN NAIVE BAYES

In [37]:
# Preparing a Gaussian naive bayes model on training data set 
from sklearn.naive_bayes import GaussianNB as GB

In [38]:
classifier_gb = GB()
classifier_gb.fit(x_train,y_train)

GaussianNB()

In [39]:
# Model Accuracy on train set
train_pred_g = classifier_gb.predict(x_train)
accuracy_train_g = np.mean(train_pred_g==y_train) 

In [40]:
# Model Accuracy on test set
test_pred_g = classifier_gb.predict(x_train)
accuracy_test_g = np.mean(test_pred_g==y_train)

In [41]:
accuracy_train_g

0.7953317197705646

In [42]:
accuracy_test_g

0.7953317197705646

In [43]:
#After applying Gaussian Naive Bayes we get 79.53% accuracy.

CONCLUSION: Categorical Naive Bayes gives us better accuracy which is about
85.71%.